morpheus/stages/general/linear_modules_stage.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import typing
+
+import srf
+
+from morpheus.config import Config
+from morpheus.pipeline.single_port_stage import SinglePortStage
+from morpheus.pipeline.stream_pair import StreamPair
+from morpheus.utils.module_utils import load_module
+
+logger = logging.getLogger("morpheus.{}".format(__name__))
+
+
+class LinearModulesStage(SinglePortStage):
```

Comment:
Lets update this Stage so that it supports user specified input and output port names from the module. We are not guaranteed that every module implementer will use 'input' and 'output' for its port names, or that a given module will be limited to sinlge inputs/outputs to select from.